### PR TITLE
[MM-16501] Delete user from Bot table when deleting user with mattermost CLI

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR addresses an issue when deleting a user via the mattermost CLI.  If the user is a bot, delete the record from the bots table.  The solution was to add the `PermanentDelete()` Bot store method inside the App's `PermanentDeleteUser` method

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16501

### CODE REVIEW

**Comment by jfrerich** [comment]
Note this does not address the suggestion from the community member in this post https://community-release.mattermost.com/core/pl/rd78be5eq3yjukgnpfpmnutjxa

File: app/user.go [hunk 1]
```diff
@@ -1466,6 +1466,10 @@ func (a *App) PermanentDeleteUser(user *model.User) *model.AppError {
 		return err
 	}
 
+	if err := a.Srv.Store.Bot().PermanentDelete(user.Id); err != nil {
+		return err
```
Comment (crspeller) [inline app/user.go]: Just to confirm. This doesn't error out if the user isn't a bot right?

Comment (jfrerich) [reply app/user.go]: @crspeller 	Correct.  I tested and the `PermanentDelete()` Bot store method returns `nil` if the user doesn't exist.

**Comment by jfrerich** [comment]
I'll write on up real quick!

**Comment by jfrerich** [comment]
Updated with unit test.  Note that I had to access the database directly. The `GetBots()` method on the bots store cannot be used to get
and compare bots from before and after the `PermanentDeleteUser() call`.  This is because `GetBots()` joins with the users table which does have the bot user removed.

**Pull request merged**